flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -397,11 +398,24 @@ protected StreamTask(
         this.throughputCalculator = environment.getThroughputMeter();
         this.bufferDebloatPeriod = getTaskConfiguration().get(BUFFER_DEBLOAT_PERIOD).toMillis();
 
-        this.bufferDebloater =
-                getTaskConfiguration().get(TaskManagerOptions.BUFFER_DEBLOAT_ENABLED)
-                        ? new BufferDebloater(
-                                getTaskConfiguration(), getEnvironment().getAllInputGates())
-                        : null;
+        if (getTaskConfiguration().get(TaskManagerOptions.BUFFER_DEBLOAT_ENABLED)) {
+            this.bufferDebloater =
+                    new BufferDebloater(
+                            getTaskConfiguration(), getEnvironment().getAllInputGates());
+            environment
+                    .getMetricGroup()
+                    .gauge(
+                            MetricNames.ESTIMATED_TIME_TO_CONSUME_BUFFERS,
+                            () ->
+                                    bufferDebloater
+                                            .getLastEstimatedTimeToConsumeBuffers()
+                                            .toMillis());
```

Comment:
Do I right remember that all metrics are collected from mailbox thread so we don't need to worry about synchronization?

Reply:
Unfortunately no. I would like it to work like that in the long term. However as it is right now, metrics are probed from external threads and the conscious decision is to have those access thread unsafe - decision was that it's better to have some infrequent incorrect metric readings rather then synchornize the code and adding huge overheads in runtime.